conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/OptimizerTimeTracker.scala

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.conformance.interpreter
+
+import org.apache.log4j.LogManager
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.functions.{col, lit}
+import za.co.absa.enceladus.utils.general.JsonUtils
+import za.co.absa.enceladus.utils.schema.SchemaUtils
+
+class OptimizerTimeTracker(inputDf: DataFrame, isWorkaroundEnabled: Boolean)(implicit spark: SparkSession) {
+  private val log = LogManager.getLogger(this.getClass)
+
+  private val maxToleratedPlanGenerationPerRuleMs = 100L
+  private val initialElapsedTimeBaselineMs = 300L
+  private var baselineTimeMs = initialElapsedTimeBaselineMs
+  private var lastExecutionPlanOptimizatioTime = 0L
+
+  private val idField1 = SchemaUtils.getUniqueName("tmpId", Option(inputDf.schema))
+  private val idField2 = s"${idField1}_2"
+  private val dfWithId = inputDf.withColumn(idField1, lit(1))
+  private val dfJustId = JsonUtils.getDataFrameFromJson(spark, Seq(s"""{"$idField2":1}""")).cache
```

Comment:
Does it broadcast automatically?

Reply:
Yes, because the table is so small the broadcasting join strategy is used.